src/app/extensions/quoting/store/quote-request/quote-request.selectors.ts

```diff
@@ -4,79 +4,69 @@ import { isEqual } from 'lodash-es';
 import { getProductEntities } from 'ish-core/store/shopping/products';
 
 import { QuoteRequestHelper } from '../../models/quote-request/quote-request.helper';
+import { QuoteRequestData } from '../../models/quote-request/quote-request.interface';
 import { getQuotingState } from '../quoting-store';
 
-import { initialState } from './quote-request.reducer';
+import { initialState, quoteRequestAdapter } from './quote-request.reducer';
 
 const getQuoteRequestState = createSelector(
   getQuotingState,
   state => (state ? state.quoteRequest : initialState)
 );
 
+const { selectAll, selectEntities } = quoteRequestAdapter.getSelectors(getQuoteRequestState);
+
 export const getSelectedQuoteRequestId = createSelector(
   getQuoteRequestState,
   state => state.selected
 );
 
+export const getSelectedQuoteRequest = createSelector(
+  selectEntities,
+  getSelectedQuoteRequestId,
+  (entities, id) => id && addStateToQuoteRequest(entities[id])
+);
+
 export const getCurrentQuoteRequests = createSelector(
-  getQuoteRequestState,
-  state =>
-    state.quoteRequests.map(item => ({
-      ...item,
-      state: QuoteRequestHelper.getQuoteRequestState(item),
-    }))
+  selectAll,
+  quoteRequests => quoteRequests.map(addStateToQuoteRequest)
 );
 
-export const getQuoteRequstItems = createSelector(
+export const getQuoteRequestItems = createSelector(
   getQuoteRequestState,
   state => state.quoteRequestItems
 );
 
-export const getActiveQuoteRequest = createSelector(
-  createSelector(
-    getCurrentQuoteRequests,
-    quoteRequests => quoteRequests.filter(item => item.editable).pop() || undefined
-  ),
-  getQuoteRequstItems,
+export const getQuoteRequestItemsWithProducts = createSelector(
+  getQuoteRequestItems,
   getProductEntities,
-  (quoteRequest, quoteRequestItems, products) =>
-    !quoteRequest
-      ? undefined
-      : {
-          ...quoteRequest,
-          state: QuoteRequestHelper.getQuoteRequestState(quoteRequest),
-          items: quoteRequestItems.map(item => ({
-            ...item,
-            product: item.productSKU ? products[item.productSKU] : undefined,
-          })),
-        }
+  (items, products) =>
+    items.map(item => ({
+      ...item,
+      product: item.productSKU ? products[item.productSKU] : undefined,
+    }))
+);
+
+export const getActiveQuoteRequest = createSelector(
+  getCurrentQuoteRequests,
+  quoteRequests => {
+    const quoteRequest = quoteRequests.find(item => item.editable);
```

Comment:
I would not move the active/editable info to the state (e.g. as a `activeQuoteRequestId`) because this would be derived state. Is it possible that more than 1 quote request is editable?

If there is only 1 at a time, we could move this: Remove the `editable` flag from the QR and introduce a state prop `activeQuoteRequestId`.
Not sure...

Reply:
more than one quote requests are possibly editable. i will change this line to get the last hit.